ensign.go

```diff
@@ -170,6 +171,27 @@ func (c *Client) Subscribe(ctx context.Context, topics ...string) (_ Subscriber,
 	return sub, nil
 }
 
+// WithCallOptions configures the next client Call to use the specified call options,
+// after the call, the call options are removed. This method returns the Client pointer
+// so that you can easily chain a call e.g. client.WithCallOptions(opts...).ListTopics()
+// -- this ensures that we don't have to pass call options in to each individual call.
+// Ensure that the clone of the client is discarded and garbage collected after use;
+// the clone cannot be used to close the connection or fetch the options.
+//
+// Experimental: call options and thread-safe cloning is an experimental feature and its
+// signature may be subject to change in the future.
+func (c *Client) WithCallOptions(opts ...grpc.CallOption) *Client {
+	// Return a clone of the client with the api interface and the opts but do not
+	// include the grpc connection to ensure only the original client can close it.
```

Comment:
Makes sense to me, maybe we should make the experimental warning more noticeable in the docstring?